// swagger-smktest/src/services/swaggerSmktest/main.js
(function (exports, require, module, __filename, __dirname) { const _91b‍=exports;_91b‍.r((function *(exports,require){"use strict";let chalk,fs,ncp,path,promisify,Listr,UpdaterRenderer,configSmktest,generateCasesSwagger,getTrainingSwaggerSmktest,validUrl;_91b‍.x([["createProject",()=>createProject],["createConfigFileTask",()=>createConfigFileTask],["runMultiTasks",()=>runMultiTasks]]);_91b‍.w("chalk",[["default",["chalk"],function(v){chalk=v}]]);_91b‍.w("fs",[["default",["fs"],function(v){fs=v}]]);_91b‍.w("ncp",[["default",["ncp"],function(v){ncp=v}]]);_91b‍.w("path",[["default",["path"],function(v){path=v}]]);_91b‍.w("util",[["promisify",["promisify"],function(v){promisify=v}]]);_91b‍.w("listr",[["default",["Listr"],function(v){Listr=v}]]);_91b‍.w("listr-update-renderer",[["default",["UpdaterRenderer"],function(v){UpdaterRenderer=v}]]);_91b‍.w("./services/createConfigFile",[["configSmktest",["configSmktest"],function(v){configSmktest=v}]]);_91b‍.w("./services/smktestSwagger",[["generateCasesSwagger",["generateCasesSwagger"],function(v){generateCasesSwagger=v}],["getTrainingSwaggerSmktest",["getTrainingSwaggerSmktest"],function(v){getTrainingSwaggerSmktest=v}]]);_91b‍.w("valid-url",[["default",["validUrl"],function(v){validUrl=v}]]);yield;_91b‍.s();















const access = promisify(fs.access);
const copy = promisify(ncp);

async function copyTemplateFiles(options) {
  return copy(options.templateDirectory, options.targetDirectory, {
    clobber: false,
  });
}

       async function createProject(options) {
  options = {
    ...options,
    targetDirectory: options.targetDirector || process.cwd,
  };

  const currentFileUrl = _91b‍._.meta.url;
  const templateDir = path.resolve(
    new URL(currentFileUrl).pathname,
    '../../templates',
    options.template.toLowerCase()
  );
  options.templateDirectory = templateDir;

  try {
    await access(templateDir, fs.constants.R_OK);
  } catch (err) {
    _91b‍.g.console.log('%s Invalid template name', chalk.red.bold('ERROR'));
  }

  await copyTemplateFiles(options);

  _91b‍.g.console.log(' %s Project ready', chalk.green.bold('DONE'));

  return true;
}

       async function createConfigFileTask(options) {
  //! Create configuration file:
  let listOfTask, tasks;

  listOfTask = [
    {
      title: 'Create ConfigFile',
      task: () => configSmktest(options),
    },
  ];

  tasks = new Listr(listOfTask, {
    renderer: UpdaterRenderer,
    collapse: false,
    showSubtasks: true,
    clearOutput: false,
  });

  await tasks.run();

  return options;
}

       async function runMultiTasks(options) {
  return createConfigFileTask(options)
    .then((options) => {
      //Options:
      let swaggerUrl = options.swaggerUrl || 'N';
      options.swaggerUrl = swaggerUrl;
      let curlLogin = options.curlLogin || 'N';
      options.curlLogin = curlLogin;
      //
      return options;
    })
    .then(async (options) => {
      //Print message for create test cases:

      if (options.swaggerBasic) {
        options.swaggerBasic = true;
        let listOfTask = [
          {
            title: 'Create Swagger/OpenApi test',
            task: () => {},
          },
        ];

        let tasks = new Listr(listOfTask);

        await tasks.run();
      }

      return options;
    })
    .then(async (options) => {
      //! Create tests suite for swagger:

      if (options.swaggerUrl !== 'N') {
        await generateCasesSwagger(options);
      }

      return options;
    })
    // .then(async (options) => {
    //   //! Print message of created test:
    //   if (options.swaggerUrl !== 'N') {
    //     let listOfTask = [
    //       {
    //         title:
    //           'Smoke tests cases generated for Swagger/OpenApi basic level',
    //         task: () => {},
    //       },
    //     ];

    //     let tasks = new Listr(listOfTask);

    //     await tasks.run();
    //   }
    //   return options;
    // })
    .then(async (options) => {
      // Apply method for get data with auth

      if (options.basicWithAuth) {
        options = await generateCasesSwagger(options);
      }
      return options;
    });
}

}))
});
